src/components/FormInput/FormInput.types.ts

```diff
@@ -0,0 +1,26 @@
+import type { TextFieldProps } from '@mui/material/TextField';
+
+export type FormInputVariant = 'outlined' | 'filled';
+export type FormInputType = 'text' | 'email' | 'password';
+export type FormInputSize = 'small' | 'medium' | 'large';
+
+export interface FormInputProps extends Omit<TextFieldProps, 'variant' | 'size'> {
+  variant?: FormInputVariant;
+  type?: FormInputType;
+  size?: FormInputSize;
+  placeholder?: string;
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  errorMessage?: string;
+  disabled?: boolean;
+  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
+  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
+  'data-testid'?: string;
+}
+
+export interface StyledFormInputProps {
+  $variant: FormInputVariant;
+  $size: FormInputSize;
+  $error: boolean;
+  $shape?: keyof typeof SHAPE_CONFIGS;
```

Comment:
StyledFormInputProps references SHAPE_CONFIGS in the $shape property (keyof typeof SHAPE_CONFIGS) but SHAPE_CONFIGS is not defined in this file or in the diffs. This creates a missing symbol and will cause a type error.

**Sugerencia:**
```typescript
Either define SHAPE_CONFIGS in this file (or import it from the constants file) or remove the $shape prop if it's not implemented yet. Example:

// if shape support is intended
export const SHAPE_CONFIGS = { /* ... */ } as const;

or change the type to: $shape?: string; // or a union of allowed shapes
```